### DISCUSSION TITLE
How to get map coordinates with only state / country / county and no city / street / etc.

### DISCUSSION THREAD

**Discussion by apcloic** [YetiForceCompany/YetiForceCRM#16282] [How to] [answered]
Hi,

I'm using Map module (configured to use Nominatim) to get geographical account coordinates.
For some accounts whose only have state / country / county informations and no city / street / post code, I can't get the map/coordinates to work.
However, if I search with state name directy on [Nominatim](https://nominatim.openstreetmap.org/), it works perfectly and return the map so I suppose it should also work on YF.
After inspecting the code in OpenStreetMapHandler.php, it looks like a row is only added to u_yf_openstreetmap_record_updater table if city field is updated or filled.
Is there any reason to limit coordinates search to records which have city field not empty ?
Did I missed something ? 

Regards,

**Comment by apcloic** [comment, marked as answer]
So basically I've just made the following change in custom/modules/OpenStreetMap/handlers/OpenStreetMapHandler.php and it works as expected, map coordinates are generated using only state/county/country.

````
		foreach (\App\Map\Coordinates::TYPE_ADDRES as $typeAddress) {
//			if (!$recordModel->isEmpty('addresslevel5' . $typeAddress) && ($recordModel->isNew() || \in_array($typeAddress, $typeAddressToUpdate))) {
			if (!$recordModel->isEmpty('addresslevel2' . $typeAddress) && ($recordModel->isNew() || \in_array($typeAddress, $typeAddressToUpdate))) {

````

**Comment by mariuszkrzaczkowski** [comment]
[Improved the handler from getting coordinates to the map](https://github.com/YetiForceCompany/YetiForceCRM/commit/7c3b2c4610259f5f8f80191a3b80a10cfd23e629)
